srtcore/core.cpp

```diff
@@ -7569,6 +7569,11 @@ void srt::CUDT::bstats(CBytePerfMon *perf, bool clear, bool instantaneous)
         perf->pktRcvUndecryptTotal  = m_stats.rcvr.undecrypted.total.count();
         perf->byteRcvUndecryptTotal = m_stats.rcvr.undecrypted.total.bytes();
 
+        
+        // Average values management
+        perf->mbpsSendRate = Bps2Mbps(m_stats.sndr.getAverageValue());
+        perf->mbpsRecvRate = Bps2Mbps(m_stats.rcvr.getAverageValue());
+
         // TODO: The following class members must be protected with a different mutex, not the m_StatsLock.
         const double interval     = (double) count_microseconds(currtime - m_stats.tsLastSampleTime);
         perf->mbpsSendRate        = double(perf->byteSent) * 8.0 / interval;
```

Comment:
This looks like a bug. Old way of calculating the rate overwrites the new value.
```c++
perf->mbpsSendRate = double(perf->byteSent) * 8.0 / interval;
```

Reply:
Before it was this becasue we needed an interval, but now as we already get an average value in Bps, we just need to convert it to Mbps. I have found this methods that is doing *8 / 1 000 000 and the results were good. But I found an issue in the filling of the table on my side (method call disappears), so I'll put it again and test this case.

Reply:
I mean, look at the whole block.

```c++
        // Average values management
        perf->mbpsSendRate = Bps2Mbps(m_stats.sndr.getAverageValue());
        perf->mbpsRecvRate = Bps2Mbps(m_stats.rcvr.getAverageValue());

        // TODO: The following class members must be protected with a different mutex, not the m_StatsLock.
        const double interval     = (double) count_microseconds(currtime - m_stats.tsLastSampleTime);
        perf->mbpsSendRate        = double(perf->byteSent) * 8.0 / interval;
```